accord-core/src/main/java/accord/messages/ReadData.java

```diff
@@ -153,7 +153,11 @@ void read(SafeCommandStore safeStore, Timestamp executeAt, PartialTxn txn)
             {
                 // TODO (expected, exceptions): should send exception to client, and consistency handle/propagate locally
                 logger.trace("{}: read failed for {}: {}", txnId, unsafeStore, throwable);
-                node.reply(replyTo, replyContext, ReadNack.Error);
+                synchronized (ReadData.this)
```

Comment:
why this lock?  all impl delegate to `CommandStores` so wouldn't need locking.

Reply:
You are right it probably isn't necessary. I reflexively added it because the existing call to `cancel` is from a `synchronized` method, but the actual implementations of cancel don't require it to be `synchronized` and are probably also idempotent so it's fine if multiple responses and cancellations happen at once. I'll remove it.